st2actions/tests/unit/test_paramiko_ssh.py

```diff
@@ -802,3 +802,35 @@ def test_use_ssh_config_port_value_provided_in_the_config(self, mock_sshclient):
 
         call_kwargs = mock_client.connect.call_args[1]
         self.assertEqual(call_kwargs['port'], 9999)
+
+    @patch('paramiko.SSHClient', Mock)
+    @patch.object(ParamikoSSHClient, '_is_key_file_needs_passphrase',
+                  MagicMock(return_value=False))
+
+    def test_socket_closed(self):
+        conn_params = {'hostname': 'dummy.host.org',
+                       'username': 'ubuntu',
+                       'password': 'pass',
+                       'timeout': '600'}
+        ssh_client = ParamikoSSHClient(**conn_params)
+        ssh_client.connect()
+
+        # Make sure .close() doesn't actually call anything real
+        ssh_client.client = Mock()
+        ssh_client.sftp_client = None
+        ssh_client.bastion_client = None
+
+        ssh_client.socket = Mock()
+
+        # Make sure we havent called any close methods at this point
+        self.assertEqual(ssh_client.socket.process.kill.call_count, 0)
```

Comment:
The mock library in Python 3.6 has built-in assertions that are probably better than `assertEqual`. Can you add a TODO comment to this 

```suggestion
        # TODO: Replace these with .assert_not_called() once it's Python 3.6+ only
        self.assertEqual(ssh_client.socket.process.kill.call_count, 0)
```